vmm/src/cpu.rs

```diff
@@ -773,7 +786,18 @@ impl CpuManager {
 
         // Only create vCPUs in excess of all the allocated vCPUs.
         for cpu_id in self.vcpus.len() as u8..desired_vcpus {
-            self.create_vcpu(cpu_id, entry_point, None)?;
+            self.create_vcpu(cpu_id)?;
+        }
+
+        Ok(())
+    }
+
+    fn configure_vcpus(&mut self, entry_point: Option<EntryPoint>) -> Result<()> {
+        for vcpu in self.vcpus.clone() {
+            if vcpu.lock().unwrap().configured {
```

Comment:
This line (or more generally the fact that we lock the Vcpu mutex) causes a deadlock when trying to hotplug new vCPU(s). I can't figure out how there's another thread that would be locking the Vcpu mutex concurrently...